hack/lib/init.sh

```diff
@@ -31,8 +31,9 @@ unset CDPATH
 # they can explicitly set GO111MODULE=on
 export GO111MODULE=off
 
-if [[ "${KUBE_PROVIDERLESS:-"off"}" == "on" ]]; then
-  export GOFLAGS=${GOFLAGS:-"-tags=providerless"}
+if [[ "${KUBE_PROVIDERLESS:-"n"}" == "y" ]]; then
+  gotags="providerless,$(echo "${GOFLAGS:-}" | sed -ne 's|.*-tags=\([^-]*\).*|\1|p')"
+  export GOFLAGS="-tags=$gotags"
```

Comment:
I think we want something like `GOFLAGS="${GOFLAGS:+${GOFLAGS} }-tags=$gotags"` ? 
So we keep other GOFLAGS but update -tags?

(bonus points for dropping the existing `-tags` but I think we can just set it again with the full tags at the end?)

Reply:
Let me do that in a follow up @BenTheElder

Reply:
/lgtm